tasks/sast-go.yaml

```diff
@@ -21,31 +21,37 @@ spec:
           memory: 512Mi
           cpu: 10m
       script: |
-        /usr/local/go/bin/gosec -no-fail -fmt=sarif -out=gosec_output.json $(workspaces.workspace.path)/... | tee gosec_output.txt
+        /usr/local/go/bin/gosec -no-fail -fmt=sarif -out=gosec_output.json $(workspaces.workspace.path)/... 2> gosec_output.txt
 
         # Test if any package was found
         # Even with -no-fail, gosec uses exit code 1 for several states,
         # including when there are no packages found.
-        grep "No packages found$" gosec_output.txt && echo "Skipping because no Go packages were found"
+        SKIP_MSG="No packages found"
+        grep -q "$SKIP_MSG" gosec_output.txt
```

Comment:
you are not changing shebang so it's using default `#!/bin/bash -xe`, so when this grep fails then it stops execution.

Reply:
Yes. Updated the scripts.

Reply:
@Michkov , could please help to review it again?

Reply:
@Michkov , could you please help to see the e2e-tests issue?